github-get-reviewers.go

```diff
@@ -33,38 +30,71 @@ type ReviewState struct {
 	State string `json:"state"`
 }
 
-var rStatus []ReviewStatus
-var rReviews []ReviewState
+func getReviewStatesForPR(repo string, number int) (error, []ReviewState) {
+	reviews := make(map[string][]github.PullRequestReview)
 
-func getReviewers(repo string, number int) []ReviewStatus {
-	rStatus = nil
 	// List all reviewers to an assigned PR # in a repo.
-	rList, _, err := buzzClient.PullRequests.ListReviewers(ctx, "minio", repo, number, nil)
+	reviewers, _, err := buzzClient.PullRequests.ListReviewers(ctx, "minio", repo, number, nil)
 	if err != nil {
-		// We should not exit on error in this case.
-		fmt.Println("Unable to get Reviewers for %s", number)
+		return err, nil
 	}
-	for _, elem := range rList {
-		eachReviewer := ReviewStatus{}
-		eachReviewer.User.Login = elem.GetLogin()
-		rStatus = append(rStatus, eachReviewer)
+
+	for _, elem := range reviewers {
+		review := github.PullRequestReview{}
+		review.User = elem
+
+		// Any reviews are later than this assignment, so put it farthest back.
+		assignDate := time.Unix(0, 0)
+		review.SubmittedAt = &assignDate
+
+		// Assume their review is pending until further notice.
+		status := "PENDING"
+		review.State = &status
+
+		reviews[elem.GetLogin()] = []github.PullRequestReview{review}
 	}
-	return rStatus
-}
 
-func getReviewStatesForPR(repo string, number int) []ReviewState {
-	rReviews = nil
-	// List all reviewers to an assigned PR # in a repo.
-	rList, _, err := buzzClient.PullRequests.ListReviews(ctx, "minio", repo, number, nil)
+	// List all who voluntarily gave a review.
+	voluntaryReviews, _, err := buzzClient.PullRequests.ListReviews(ctx, "minio", repo, number, nil)
 	if err != nil {
-		// We should not exit on error in this case.
-		fmt.Println("Unable to get Reviewers for %s", number)
+		return err, nil
+	}
+
+	for _, elem := range voluntaryReviews {
+		// Ignore comments. If a user was not assigned but has given a review anyways,
+		// add their review as new. If they've been assigned previously and are
+		// continuing with their review, the new one is appended.
+		switch elem.GetState() {
+		case "COMMENTED":
+			continue
+		default:
+			username := elem.User.GetLogin()
+
+			if _, ok := reviews[username]; ok {
+				reviews[username] = append(reviews[username], *elem)
```

Comment:
I guess now that i see it can be made even simpler by just doing 

```
if elem.GetState() != "COMMENTED" {
        username := elem.User.GetLogin()
        if _, ok := reviews[username]; ok {
                   reviews[username] = append(reviews[username], *elem)
        } else {
                   reviews[username] = []github.PullRequestReview{*elem}
        }
}